src/kzg.rs

```diff
@@ -390,7 +390,11 @@ impl Kzg {
     }
 
     /// commit the actual polynomial with the values setup
-    pub fn commit(&self, polynomial: &Polynomial) -> Result<G1Affine, KzgError> {
+    pub fn commit(
+        &self,
+        polynomial: &Polynomial,
+        is_data_polynomial_evaluations: bool,
```

Comment:
thinking if making this an enum with values like `in_coefficient_form` or 'in_eval_form' to make it clearer. open to other options.

Reply:
I think that enum would make it more clear, will make that change

Reply:
https://github.com/Layr-Labs/rust-kzg-bn254/commit/b26c93426bb1981fb1a81daf2c3468188df521de

Took a slightly different approach here, added the enum to the Polynomial object and use that to inform whether to perform an IFFT on the g1_srs or not